lobby-client-http/src/main/java/org/triplea/http/client/error/report/ErrorUploadClient.java

```diff
@@ -1,20 +1,40 @@
 package org.triplea.http.client.error.report;
 
+import java.net.URI;
+
+import org.triplea.http.client.HttpClient;
+
 import com.google.common.annotations.VisibleForTesting;
 
+import feign.FeignException;
 import feign.Headers;
 import feign.RequestLine;
 
+/**
+ * Http client to upload error reports to the http lobby server.
+ */
 @SuppressWarnings("InterfaceNeverImplemented")
```

Comment:
Here again: Adding a comment, where the interface is actually being implemented would be nice.

Reply:
Looking a bit further down in the class, there is this method:
```
  static ErrorUploadClient newClient(final URI uri) {
    return new HttpClient<>(ErrorUploadClient.class, uri).get();
  }
```
The return type is an instance of this class, so that is the location this is instantiated. Following that can lead to where we spawn instances. But actually, yes, there is no implementation of the interface as it is auto-built.

Given the answer is in the same class, and is a bit parcel to how Feign works, I'm not sure how to document this appropriately. It might be a fundamental learning curve of how Feign works and how we use http clients. There are tests to document how to spawn new instances. I suspect if someone really wondered they could look there.

Otherwise, I would suggest we create a centralized glossary of suppressions. The copy/pasted commentary on other suppressions bugs me personally and I think would also be better in a glossary.

@RoiEXLab any thoughts/suggestions?

Reply:
That's a valid concern I haven't thought of.
What do you think about adding a line to the Javadoc of the class `Implemented in {@link #newClient()}.`?
I know that this doesn't really solve the "generic comment problem" (unless the method name differs in other classes), but at least the comment would get renamed as well when doing any refactoring.
I know this isn't the "solution" you searched for but "just having to know how feign works" is not an ideal solution either IMO.
But we should definitely document this somewhere in any case and notify the other devs about it.

Reply:
>  "just having to know how feign works"

I misspoke, the factory method hides the details and the details are easily traced, and feign is open source itself. 

I think standard "where is this interface implemented" techniques should find the answer pretty fast.  We could move the static factory method to the top of the interface as well.

@RoiEXLab I think you might be over-indexing on the suppression. I would almost just as well remove it. On the plus side all such interfaces work the same. For example, TripleA has a pattern of '*Controller'  classes for DAO. Should we document that on each and every class?

Because the method to construct an instance is a few lines away from the suppression, I really don't think it's a problem that needs to be documented.

Reply:
I'm resolving this convo, the suppression is already committed code, it's not even a part or element of this PR.